system/src/system_network_diagnostics.cpp

```diff
@@ -321,7 +328,7 @@ class NetworkCellularCellGlobalIdentityLocationAreaCodeDiagnosticData
 
     virtual int get(IntType& val)
     {
-        CellularGlobalIdentity cgi;
+        CellularGlobalIdentity cgi{sizeof(CellularGlobalIdentity),CGI_VERSION_LATEST};
```

Comment:
I actually prefer those, but I thought that was only a `C` thing. I will switch over.

Reply:
AFAIK it's in C++20 officially, but GCC supported trivial cases since long ago.

Reply:
I just did a cursory check and I cannot find where designated initialization has made it into the c++ standard.
@avtolstoy Can you please confirm when it was added to the standard and if our compile flags `-std=c++14` support it?

Reply:
See message above. Yes, it'll work as long as the designators are specified in the same order as the fields during the declaration.

Reply:
I know it will work, but I'm more curious as to whether this code is inline with the standard we are compiling our code against. While I truly like the clarity of the syntax, I would prefer to maintain standard compliance and not become compiler dependent.

Reply:
In that case you should consider removing `__attribute__` or any other GNU Extensions as well.

I simply want to improve readability, so you could also consider:
```suggestion
        CellularGlobalIdentity cgi = {};
        cgi.size = sizeof(cgi);
        cgi.version = CGI_VERSION_LATEST;
```

Reply:
You make a good point about `__attribute__`, seems like we are well committed to GNU at this point. I will follow your first suggestion, and use the designated initializers so we can keep it on one line.